content/sensu-core/1.0/guides/pre-compile-plugins.md

```diff
@@ -0,0 +1,146 @@
+---
+title: "Pre-Compiling Plugins"
+description: "Pre-Compiling Sensu Plugins for Distribution"
+product: "Sensu Core"
+version: "1.0"
+weight: 2
+next: ../intro-to-checks
+previous: ../overview
+menu:
+  sensu-core-1.0:
+    parent: guides
+---
+
+# Overview
+
+This guide will walk you through an example on how to pre-compile Sensu Plugins for distribution among clients.
+The goal of this guide is to showcase how you can quickly install plugins on ephemeral instances where time to compile can affect time to production  .
+We'll be using `sensu-plugins-aws` as it requires `nokogiri`,  usually requiring a longer build time.
+
+# Prerequist
+
+Sources:
+* [GitHub - luislavena/gem-compiler: A RubyGems plugin that generates binary gems](https://github.com/luislavena/gem-compiler)
+This guide assumes the following:
+* Centos 7 platform
+* You have some way to store these artifacts and make them available for systems in your environment
+_NOTE: Compiled gems are tied to the Ruby version they are compiled for. If Sensu upgrade Ruby 2.4 -> Ruby 2.5, packages will need to be rebuilt._
+_NOTE: Compiled gems are tied to CPU architecture for which they are compiled on._
+
+## Installing The Build Toolchain
+
+We'll first add the Sensu yum repository definition and install Sensu
+
+```bash
+printf '[sensu]
+name=sensu
+baseurl=https://repositories.sensuapp.org/yum/$releasever/$basearch/
+gpgcheck=0
+enabled=1' | sudo tee /etc/yum.repos.d/sensu.repo && yum install -y sensu
+```
+
+And then install EPEL and Dev tools
+
+```bash
+sudo yum install -y epel-release
+sudo yum groupinstall -y "Development Tools"
+```
+
+Finally we'll add Sensu's embedded bin to our path
+
+```bash
+export PATH=$PATH:/opt/sensu/embedded/bin
+```
+
+## Installing a Gem to a tmp directory
+
+Install the gem-compiler package
+
+```bash
+gem install gem-compiler
+```
+
+We'll now install the source gems into a cache directory
+
+```bash
+mkdir /tmp/gems
+gem install --no-ri --no-rdoc --install-dir /tmp/gems sensu-plugins-aws
+```
+
+Observing the `gem install` output, we can see that nokogiri and unf_ext build native extensions.
+These are the gems we will compile so we can install them from our own repository.
+
+### Compile the Gems for packaging
+
+We'll use `gem-compiler` to rebuild the gem packages with the included native extensions:
+
+```bash
+cd /tmp/gems/cache
+gem compile unf_ext-0.0.7.4.gem
```

Comment:
This command fails due to the gem being `unf_ext-0.0.7.5-x86_64-linux.gem`. I'd recommend modifying the instructions so that you're not going after a specific version, as this is going to cause a bit of headache as that version increments.

Reply:
pinned the versions in the install, this should fix the above while still being verbose.